plugins/org.komodo.shell/src/org/komodo/shell/commands/core/RenameCommand.java

```diff
@@ -78,16 +79,27 @@ public boolean execute() throws Exception {
         }
 
         try {
+            // If teiid object was renamed, check if it is set as the default server.  unset default if necessary
+            KomodoObject kObject = objContext.getKomodoObj();
```

Comment:
Maybe replacing 83-90 with just using the KomodoType to determine if a Teiid object was renamed might be better. The `WorkspaceManager.resolve` method sometimes has to walk all the object types to determine the exact type. So for instance you could rewrite like this:
```
if (KomodoType.TEIID == kType) {
	final Teiid defaultTeiid = wsStatus.getTeiid();

	if ((defaultTeiid != null) && defaultTeiid.getName(wsStatus.getTransaction()).equals(objContext.getName())) {
		wsStatus.setTeiid(null);
	}
}
```
This change could also be made in the `DeleteCommand` if you want.